zebra-chain/src/transaction/serialize.rs

```diff
@@ -377,16 +379,20 @@ impl ZcashSerialize for orchard::ShieldedData {
         // Denoted as `bindingSigOrchard` in the spec.
         self.binding_sig.zcash_serialize(&mut writer)?;
 
+        #[cfg(feature = "tx-v6")]
+        // Denoted as `vAssetBurn` in the spec (ZIP 230).
+        self.burn.zcash_serialize(&mut writer)?;
```

Comment:
Burn is dependent on `V`. 
For `OrchardVanilla::Burn` we should do nothing but must call it
For `OrchardZSA::Burn` we should serialize as expected.